gloo/docs/user_guides/tls_setup.md

```diff
@@ -0,0 +1,317 @@
+    --dest-name default-petstore-8080 \
+    --prefix-rewrite /api/pets
+```
+Since we didn't explicitly create a VirtualSerivce, adding this route will create a default VirtualService named `default`.
+
+```bash
+glooctl get virtualservice default -o yaml
+```
+
+```yaml
+---
+metadata:
+  name: default
+  namespace: gloo-system
+  resourceVersion: "21723"
+status:
+  reportedBy: gateway
+  state: Accepted
+  subresourceStatuses:
+    '*v1.Proxy gloo-system gateway-proxy':
+      reportedBy: gloo
+      state: Accepted
+virtualHost:
+  domains:
+  - '*'
+  name: gloo-system.default
+  routes:
+  - matcher:
+      exact: /sample-route-1
+    routeAction:
+      single:
+        upstream:
+          name: default-petstore-8080
+          namespace: gloo-system
+    routePlugins:
+      prefixRewrite:
+        prefixRewrite: /api/pets
+```
+
+If we want to query the service to verify routing is working, we can like this:
+
+```bash
+curl $(glooctl proxy url --port http)/sample-route-1
+[{"id":1,"name":"Dog","status":"available"},{"id":2,"name":"Cat","status":"pending"}]
+```
+
+Let's enable HTTPS by configuring TLS/SSL for our VirtualService.
+
+### Configuring TLS/SSL in a VirtualService
+
+Before we add the TLS/SSL configuration, let's create a private key and certificate to use in our VirtualService. Obviously if you have your own key/cert pair, you can use those instead of creating self-signed certs here.
+
+```bash
+openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+ -keyout tls.key -out tls.crt -subj "/CN=petstore.example.com"
+```
+
+Now we should create the Kubernetes secrets to hold this cert:
+
+```bash
+kubectl create secret tls gateway-tls --key tls.key \
+--cert tls.crt --namespace gloo-system
+```
+
+Lastly, let's configure the VirtualService to use this cert via the Kubernetes secrets:
+
+```bash
+glooctl edit virtualservice --name default --namespace gloo-system \
+--ssl-secret-name gateway-tls --ssl-secret-namespace gloo-system
+```
+
+Now if we get the `default` VirtualService, we should see the new SSL configuration:
+
+```bash
+glooctl get virtualservice default -o yaml
+```
+
+{{< highlight yaml "hl_lines=6-9" >}}
+---
+metadata:
+  name: default
+  namespace: gloo-system
+  resourceVersion: "22639"
+sslConfig:
+  secretRef:
+    name: gateway-tls
+    namespace: gloo-system
+status:
+  reportedBy: gateway
+  state: Accepted
+  subresourceStatuses:
+    '*v1.Proxy gloo-system gateway-proxy':
+      reportedBy: gloo
+      state: Accepted
+virtualHost:
+  domains:
+  - '*'
+  name: gloo-system.default
+  routes:
+  - matcher:
+      exact: /sample-route-1
+    routeAction:
+      single:
+        upstream:
+          name: default-petstore-8080
+          namespace: gloo-system
+    routePlugins:
+      prefixRewrite:
+        prefixRewrite: /api/pets
+{{< /highlight >}}
+
+If we try query the HTTP port, we should not get a successful response:
+
+```bash
+curl $(glooctl proxy url --port http)/sample-route-1
```

Comment:
I'd include error result to make this example more clear. I initially glanced and looked like you made same call twice and not clear what difference I was looking for. Had to carefully read surrounding text to understand...